### PR TITLE
Feature | Advanced QB Having Clause

### PR DESCRIPTION
**Pull request by edwinmp**
Ticket #557

### CODE REVIEW

**Comment by edwinmp** [comment]
@ThatcherK 
I created this PR so I can't really add code reviews via the files tab... anyway, this is a functional review.

- [x] Under the having conditions, I think we can have more than equal, less & greater ... maybe as many as we do for the filter
- [x] Change the `Replace` button caption to `Add` ... do this for the filter as well while you're at it.

For those that know what they're doing, this will be very useful ... great work.

**Comment by ThatcherK** [comment]
@edwinmp done with the functional feedback

**Comment by edwinmp** [comment]
@ThatcherK see the screenshot below - this is how the edit view appears for the having clause. It seems you're not doing a reverse mapping so the value is an object string.

image

2. When I try to groupby a string field & then I select the having clause, the app crashes with a massive error (probably cos it shouldn't be allowed)
3. Since your having conditions only work on numerical fields, `contains` should not be in the conditional list. They shouldn't only work on numerical fields or expressions, but I guess the most useful are those so lets leave them at that till it's requested.
4. I cannot add a having condition that isn't based on an aggregate ... e.g Amount USD > 20000 - somehow, the view forces the options on the value side to be aggregates. I guess this kind of usage would belong in a filter, but still, it's weird to force an aggregate.

File: frontend/src/components/JqueryQueryBuilder/JqueryQueryBuilder.tsx [hunk 1]
```diff
@@ -32,9 +36,38 @@ const JqueryQueryBuilder: FunctionComponent<JqueryQueryBuilderProps> = ({
...
+            (previousValue as { column: string; aggregate: string }).aggregate,
+          ).map((option) => {
+            return {
+              value: `${previousValue?.column},${option}`,
+              label: `${option}(${previousValue?.column})`,
+            };
+          });
+        }
+      }
+      console.log(rules);
```
Comment (edwinmp) [inline frontend/src/components/JqueryQueryBuilder/JqueryQueryBuilder.tsx]: @ThatcherK Please do this outside of this component... it's what props are for cos this is crowding what was initially a simple component.

**Comment by ThatcherK** [comment]
@edwinmp I need your review here

**Pull request closed without merging**